src/BoundaryConditions/boundary_condition.jl

```diff
@@ -87,6 +87,7 @@ const HBC  = BoundaryCondition{<:Halo}
     PeriodicBoundaryCondition() = BoundaryCondition(Periodic, nothing)
       NoFluxBoundaryCondition() = BoundaryCondition(Flux,     nothing)
 ImpenetrableBoundaryCondition() = BoundaryCondition(Open,     nothing)
+        HaloBoundaryCondition() = BoundaryCondition(Halo,     nothing)
```

Comment:
`CommunicationBoundaryCondition`? (All boundary conditions _involve_ the halos, so `HaloBoundaryCondition` is a little ambiguous)

Reply:
That can work!